twingraph/src/main/kotlin/com/cosmotech/twingraph/api/TwingraphServiceImpl.kt

```diff
@@ -80,26 +80,12 @@ class TwingraphServiceImpl(
 
   @Suppress("SpreadOperator")
   override fun delete(organizationId: String, graphId: String) {
-    val organization = organizationService.findOrganizationById(organizationId)
-    csmRbac.verify(organization.getRbac(), PERMISSION_READ)
-
-    csmJedisPool.resource.use { jedis ->
-      val matchingKeys = mutableSetOf<String>()
-      var nextCursor = ScanParams.SCAN_POINTER_START
-      do {
-        val scanResult = jedis.scan(nextCursor, ScanParams().match("$graphId*"))
-        nextCursor = scanResult.cursor
-        matchingKeys.addAll(scanResult.result)
-      } while (!nextCursor.equals(ScanParams.SCAN_POINTER_START))
-
-      val count = jedis.del(*matchingKeys.toTypedArray())
-      logger.debug("$count keys are removed from Twingraph with prefix $graphId")
+    for (item in getVersions(graphId)) {
```

Comment:
Better to do it in functional way: 
```
getVersions(graphId).forEach {
csmRedisGraph.deleteGraph(it)
}
```
Before, the code you dropped, it was even better `jedis.del(*matchingKeys.toTypedArray())`  because one Redis operator is better than multiple operator in loop.